MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/database/AsyncDatabaseExecutor.java

```diff
@@ -0,0 +1,46 @@
+package com.google.moviestvsentiments.service.database;
+
+import static android.os.Process.THREAD_PRIORITY_BACKGROUND;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * An Executor that executes all queries on background threads.
+ */
+public class AsyncDatabaseExecutor implements Executor {
+
+    private final ExecutorService executor;
+
+    /**
+     * Creates an AsyncDatabaseExecutor object.
+     */
+    private AsyncDatabaseExecutor() {
+        executor = Executors.newCachedThreadPool(new ThreadFactory() {
+            @Override
+            public Thread newThread(Runnable r) {
+                Thread thread = new Thread(r);
+                thread.setPriority(THREAD_PRIORITY_BACKGROUND);
```

Comment:
THREAD_PRIORITY_DEFAULT should be used with setThreadPriority(int) per documentation.

https://stackoverflow.com/questions/5198518/whats-the-difference-between-thread-setpriority-and-android-os-process-setthr

Reply:
Sorry, should be THREAD_PRIORITY_BACKGROUND.